package/base-files/files/bin/board_detect

```diff
@@ -1,14 +1,20 @@
 #!/bin/sh
 
-CFG=$1
+CFG="${1:-/etc/board.json}"
 
-[ -n "$CFG" ] || CFG=/etc/board.json
-
-[ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
-		$(. $a)
+if [ -d '/etc/board.d/' ] && \
+   [ ! -s "${CFG}" ]; then
```

Comment:
Can we drop those superfluous curly braces? They just add noise. IMHO we should only do curly braces if a variable is adjacent to alphanumerics (e.g. `echo "${num}min"`) or using substitution (e.g. `echo "${name%.ext}"`)

Reply:
Adding curly braces is considered good shell scripting practice. Reason being a few. For one, consistency, always put curly-brackets around variables. More importantly however, don't force the reader to learn all intricacies of what a variable is. By simply following this rule, one finally also avoids bugs, which is always a win.

As for noise, that is up to the reader, if one dislikes reading the curly brackets, it feels like noise, if one is accustomed to them, not having them feels amiss :)